libs/sdk-ui-ext/src/internal/components/pluggableVisualizations/comboChart/PluggableComboChart.tsx

```diff
@@ -235,6 +244,76 @@ export class PluggableComboChart extends PluggableBaseChart {
         );
     }
 
+    protected getDefaultAndAvailableSort(
+        buckets: IBucketOfFun[],
+        properties: IVisualizationProperties,
+    ): {
+        defaultSort: ISortConfig["currentSort"];
+        availableSorts: ISortConfig["availableSorts"];
+    } {
+        const measures = getBucketItemsByType(buckets, BucketNames.MEASURES, [METRIC]);
+        const secondaryMeasures = getBucketItemsByType(buckets, BucketNames.SECONDARY_MEASURES, [METRIC]);
+        const viewBy = getBucketItems(buckets, BucketNames.VIEW);
+        const stackMeasures = properties?.controls?.stackMeasures;
+        const defaultSort = viewBy.map((vb) => newAttributeSort(vb.localIdentifier, "asc"));
+
+        if (!isEmpty(viewBy) && (!isEmpty(measures) || !isEmpty(secondaryMeasures))) {
+            const mergedMeasures = [...measures, ...secondaryMeasures];
+            const areaSortEnabled = mergedMeasures.length > 1 ? true : false;
+
+            if (stackMeasures) {
+                return {
+                    defaultSort,
+                    availableSorts: [
+                        {
+                            itemId: localIdRef(viewBy[0].localIdentifier),
+                            attributeSort: {
+                                areaSortEnabled: true,
+                                normalSortEnabled: true,
+                            },
+                        },
+                    ],
+                };
+            }
+
+            return {
+                defaultSort,
+                availableSorts: [
+                    {
+                        itemId: localIdRef(viewBy[0].localIdentifier),
+                        attributeSort: {
+                            areaSortEnabled,
+                            normalSortEnabled: true,
+                        },
+                        metricSorts: [
+                            ...mergedMeasures.map((m) => newMeasureSortSuggestion(m.localIdentifier)),
+                        ],
+                    },
+                ],
+            };
+        }
+
+        return {
+            defaultSort: [],
+            availableSorts: [],
+        };
+    }
+
+    public getSortConfig(referencePoint: IReferencePoint): Promise<ISortConfig> {
+        const { buckets, properties } = referencePoint;
+        const measures = getBucketItemsByType(buckets, BucketNames.MEASURES, [METRIC]);
+        const secondaryMeasures = getBucketItemsByType(buckets, BucketNames.SECONDARY_MEASURES, [METRIC]);
+        const viewBy = getBucketItems(buckets, BucketNames.VIEW);
+        const disabled = viewBy.length < 1 || (measures.length < 1 && secondaryMeasures.length < 1);
```

Comment:
Please extend this condition with `... || availableSorts.length === 0;`.
This requires to switch line 307 with line 308. I would want returned sort config to be in correct state and remove condition from AD that checks `availableSorts` length and disables button if it is empty.